sdk/core/src/request_options/timeout.rs

```diff
@@ -12,6 +12,10 @@ impl Timeout {
 
 impl AppendToUrlQuery for Timeout {
     fn append_to_url_query(&self, url: &mut url::Url) {
+        if url.query_pairs().any(|(k, _)| k == "timeout") {
+            return;
+        }
```

Comment:
Wouldn't it make more sense to update the value if it is present? You could short-circuit string reallocation if it's the same value.

Reply:
As far as I can tell from the API for URL, you can't update the query pairs, only add.